parser/testdata/workflows/greetings-v08-spec.sw.yaml

```diff
@@ -170,6 +170,7 @@ states:
     type: inject
     data:
       result: Hello World, last state!
+      boolValue: false
```

Comment:
Can you also add the test case where originated this problem?

Reply:
Not sure if I follow, this boolean example has caused the issue.
What else would you like to see in the new test case?

Reply:
The SW that originated this problem. Wasn't the `compensate: true`, the problem?

Reply:
No, only this field, as the object struct was dealing only with string, raw or int, and int was the default, thus we had the int32 error in the message.

the states transition.compensate is correctly being marshaled/unmarshaled.

but I see your point, will add a test case with the entire flow.